utilization.md

```diff
@@ -1,142 +1,196 @@
+	5. [Mirror view](#mirror-view)
+		1. [Always create and allocate](#always-create-and-allocate)
+		2. [Create and allocate if source view is not in host space](#create-and-allocate-if-source-view-is-not-in-host-space)
+		3. [Create, allocate and synchronize if source view is not in same space as destination view](#create-allocate-and-synchronize-if-source-view-is-not-in-same-space-as-destination-view)
+	6. [Subview](#subview)
+	7. [Scatter view (experimental)](#scatter-view-experimental)
+		1. [Specific header](#specific-header)
+		2. [Create](#create)
+		3. [Scatter operation](#scatter-operation)
+		4. [Scatter](#scatter)
+		5. [Compute](#compute)
+		6. [Gather](#gather)
+8. [Parallelism patterns](#parallelism-patterns)
+	1. [For loop](#for-loop)
+	2. [Reduction](#reduction)
+	3. [Fences](#fences)
+		1. [Global fence](#global-fence)
+		2. [Execution space fence](#execution-space-fence)
+		3. [Team barrier](#team-barrier)
+4. [Execution policy](#execution-policy)
+	1. [Create](#create)
+	2. [Ranges](#ranges)
+		1. [One-dimensional range](#one-dimensional-range)
+		2. [Multi-dimensional (dimension 2)](#multi-dimensional-dimension-2)
+	3. [Hierarchical parallelism](#hierarchical-parallelism)
+		1. [Team policy](#team-policy)
+		2. [Team vector level (2-level hierarchy)](#team-vector-level-2-level-hierarchy)
+			1. [One-dimensional range](#one-dimensional-range)
+			2. [Multi-dimensional range (dimension 2)](#multi-dimensional-range-dimension-2)
+		3. [Team thread vector level (3-level hierarchy)](#team-thread-vector-level-3-level-hierarchy)
+			1. [One-dimensional range](#one-dimensional-range)
+			2. [Multi-dimensional range (dimension 2)](#multi-dimensional-range-dimension-2)
+4. [Scratch memory](#scratch-memory)
+	1. [Scratch memory space](#scratch-memory-space)
+	2. [Create and populate](#create-and-populate)
+5. [Atomics](#atomics)
+	1. [Atomic operations](#atomic-operations)
+	2. [Atomic exchanges](#atomic-exchanges)
+6. [Mathematics](#mathematics)
+	1. [Math functions](#math-functions)
+	2. [Complex numbers](#complex-numbers)
+		1. [Create](#create)
+		2. [Manage](#manage)
+3. [Utilities](#utilities)
+	1. [Code interruption](#code-interruption)
+	2. [Print inside a kernel](#print-inside-a-kernel)
+	3. [Timer](#timer)
+		1. [Create](#create)
+		2. [Manage](#manage)
+	4. [Manage parallel environment](#manage-parallel-environment)
+4. [Macros](#macros)
+	1. [Essential macros](#essential-macros)
+	2. [Extra macros](#extra-macros)
 
 <!--#endif-->
 
+## Header
+
+```cpp
+#include <Kokkos_Core.hpp>
+```
+
 ## Initialization
 
-### Headers
+### Initialize and finalize
 
 ```cpp
-#include <Kokkos_Core.hpp>
+int main(int argc, char* argv[]) {
+    Kokkos::initialize();
+    { /* ... */ }
+    Kokkos::finalize();
+    return 0;
+}
 ```
 
-### Initialize and finalize Kokkos
+### Scope guard
 
 ```cpp
-Kokkos::initialize(int& argc, char* argv[]);
-{
+int main(int argc, char* argv[]) {
+    Kokkos::ScopeGuard kokkos();
     /* ... */
+    return 0;
```

Comment:
Still not sure if it worth mentioning it.

Reply:
Ok, let's comment it for now before this feature is tested.